### ISSUE TITLE
Webpack is recognized build 404 path as 404.html/ instead of 404/

### ISSUE THREAD

**Issue by rayriffy** [gatsbyjs/gatsby#16087]
## Description

![Console](https://cdn.discordapp.com/attachments/577046038716874762/603913456797745152/image0.png)

Webpack cannot find `public/page-data/404.html/page-data.json` because the real path is `public/page-data/404/page-data.json`

I also got a Travis log for you here https://travis-ci.org/rayriffy/rayriffy-blog/builds/563508880#L370

### Steps to reproduce

1. Clone `master` branch of https://github.com/rayriffy/rayriffy-blog
2. Execute `yarn build` for first time.
3. `yarn build` again

### Expected result

Builds after the first build should be passed

### Actual result

It shouldn't

### Environment

```

  System:
    OS: macOS 10.15
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.6.0 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 75.0.3770.142
    Safari: 13.0
  npmPackages:
    gatsby: ^2.9.2 => 2.10.0 
    gatsby-image: ^2.2.1 => 2.2.1 
    gatsby-plugin-feed: ^2.3.0 => 2.3.0 
    gatsby-plugin-google-analytics: ^2.1.0 => 2.1.0 
    gatsby-plugin-google-fonts: ^1.0.1 => 1.0.1 
    gatsby-plugin-manifest: ^2.2.0 => 2.2.0 
    gatsby-plugin-netlify: ^2.1.0 => 2.1.0 
    gatsby-plugin-netlify-cache: ^1.2.0 => 1.2.0 
    gatsby-plugin-react-helmet: ^3.1.0 => 3.1.0 
    gatsby-plugin-remove-serviceworker: ^1.0.0 => 1.0.0 
    gatsby-plugin-robots-txt: ^1.4.0 => 1.4.0 
    gatsby-plugin-sharp: ^2.2.1 => 2.2.1 
    gatsby-plugin-sitemap: ^2.2.0 => 2.2.0 
    gatsby-plugin-styled-components: ^3.1.0 => 3.1.0 
    gatsby-plugin-typescript: ^2.1.0 => 2.1.0 
    gatsby-plugin-typography: ^2.3.0 => 2.3.0 
    gatsby-remark-copy-linked-files: ^2.1.0 => 2.1.0 
    gatsby-remark-embed-gist: ^1.1.7 => 1.1.7 
    gatsby-remark-embed-spotify: ^2.0.2 => 2.0.2 
    gatsby-remark-embed-video: ^1.7.1 => 1.7.1 
    gatsby-remark-images: ^3.0.14 => 3.1.0 
    gatsby-remark-prismjs: ^3.3.0 => 3.3.0 
    gatsby-remark-responsive-iframe: ^2.2.0 => 2.2.0 
    gatsby-remark-smartypants: ^2.1.0 => 2.1.0 
    gatsby-source-filesystem: ^2.0.39 => 2.1.0 
    gatsby-transformer-json: ^2.2.0 => 2.2.0 
    gatsby-transformer-remark: ^2.3.12 => 2.5.0 
    gatsby-transformer-sharp: ^2.2.0 => 2.2.0 

```

**Comment by pieh** [comment]
Thanks for reproduction!

It seems like this step is deleting `public/page-data/404.html/page-data.json` - https://github.com/gatsbyjs/gatsby/blob/42317c16ea1f86b1b3fe04b91cc25e16b79f757b/packages/gatsby/src/bootstrap/index.js#L123-L140

We should blacklist `page-data` from being deleted there.

This often is not causing problems when there are no queries in 404 page (or queries don't get invalidated)

**Comment by stefanprobst** [comment]
This should be fixed in gatsby@2.13.51, see #16354

**Comment by rayriffy** [comment]
Woah! Thank you very much! :)
